### ISSUE TITLE
Subnet `purpose` not respected

### ISSUE THREAD

**Issue by bcramer-c1** [terraform-google-modules/terraform-google-network#491]
### TL;DR

The `purprose` parameter within the subnet configuration is not respected. Unable to create subnets whose purpose is anything other than `PRIVATE`

### Expected behavior

The `purpose` attribute behaves like described here: https://cloud.google.com/vpc/docs/subnets#purpose

Where the applicable values could be `REGIONAL_MANAGED_PROXY`, `PRIVATE_SERVICE_CONNECT`, etc.

### Observed behavior

Every subnet is created where with the `purpose` as `PRIVATE`

### Terraform Configuration

```hcl
Module configuration


module "subnets" {
  source  = "terraform-google-modules/network/google//modules/subnets"
  version = "7.3.0"

  project_id              = var.project_id
  network_name            = var.name
  subnets                 = var.subnets
  secondary_ranges        = var.secondary_ranges
}

Variable

subnets = [
  {
    subnet_ip        = "REDACTED"
    subnet_name      = "REDACTED"
    subnet_region    = "us-central1"
    purpose          = "REGIONAL_MANAGED_PROXY"
    role             = "ACTIVE"
    description      = "Proxy only subnet for Regional Envoy-based Load Balancers in the us-central1 region"
  }
```

### Terraform Version

```sh
Terraform v1.5.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.80.0
+ provider registry.terraform.io/hashicorp/google-beta v4.80.0
```

### Additional information

This may very well affect other parameters also, like `role`.
